lib/DefinePlugin.js

```diff
@@ -507,29 +518,75 @@ class DefinePlugin {
 								"DefinePlugin",
 								withValueDependency(key, evaluateToString("object"))
 							);
-						parser.hooks.expression.for(key).tap("DefinePlugin", expr => {
-							addValueDependency(key);
-							const strCode = stringifyObj(
-								obj,
+						if (!expressionHandled) {
+							parser.hooks.expression.for(key).tap("DefinePlugin", expr => {
+								addValueDependency(key);
+								const strCode = stringifyObj(
+									obj,
+									parser,
+									compilation.valueCacheVersions,
+									key,
+									runtimeTemplate,
+									!parser.isAsiPosition(expr.range[0])
+								);
+
+								if (/__webpack_require__\s*(!?\.)/.test(strCode)) {
+									return toConstantDependency(parser, strCode, [
+										RuntimeGlobals.require
+									])(expr);
+								} else if (/__webpack_require__/.test(strCode)) {
+									return toConstantDependency(parser, strCode, [
+										RuntimeGlobals.requireScope
+									])(expr);
+								} else {
+									return toConstantDependency(parser, strCode)(expr);
+								}
+							});
+							const undefinedHandler = toConstantDependency(
 								parser,
-								compilation.valueCacheVersions,
-								key,
-								runtimeTemplate,
-								!parser.isAsiPosition(expr.range[0])
+								"undefined"
 							);
+							parser.hooks.expressionMemberChain
+								.for(key)
```

Comment:
I think `expressionMemberChain` is only called for the root object, so would this logic also work for e. g. 

```
DefinePlugin({
  "NESTED.OBJECT": { ... }
})
```

where `NESTED` is not defined.

Reply:
fixed